src/main/java/seedu/address/model/person/GroupContainsKeywordsPredicate.java

```diff
@@ -0,0 +1,58 @@
+package seedu.address.model.person;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.StringUtil;
+import seedu.address.commons.util.ToStringBuilder;
+
+/**
+ * Tests that a {@code Person}'s {@code Group} matches any of the keywords given.
+ */
+public class GroupContainsKeywordsPredicate implements Predicate<Person> {
+    private final List<String> keywords;
+
+    /**
+     * Constructor for GroupContainsKeywordsPredicate.
+     * @param keywords List of keywords to test.
+     */
+    public GroupContainsKeywordsPredicate(List<String> keywords) {
+        this.keywords = keywords;
+    }
+
+    /**
+     * Tests that a {@code Person}'s {@code Group} matches any of the keywords given.
+     * @param person Person to test.
+     * @return True if the person's group matches any of the keywords given.
+     */
+    @Override
+    public boolean test(Person person) {
+        return keywords.stream()
+                .anyMatch(keyword -> StringUtil.containsWordIgnoreCase(person.getGroup().value, keyword));
+    }
+
```

Comment:
I like the use of functional programming! It makes for better readability and maintainability!